oop-special_class_method-medium-q9 (#647)
canvas-gamification/canvas-gamification-junit-tests: src/oop/special_class_method/medium/q9/QuestionText.md

# oop-special_class_method-medium-q9

> 

Reviewer: mark either in here on the Main Test what kind of question this is

## Question Text

Suppose you are an employee of a vehicle manufacturing company, and you want to impress your boss with your Java skills. 
Make a Java class called **Car** with two attributes called **maker** and **topSpeed** that are initialized by 

> the constructor. You will need to compose the necessary methods so that these attributes can be accessed from outside 

Reviewer: ```suggestion
the constructor. The necessary methods must be set up so that these attributes can be accessed from outside 
```

of the class but they cannot be modified.

### Car Class:

> Your `Car` class must have the following items

Reviewer: modify this to express that the items below are provided

- Attributes:
    - An `String` called **maker**
    - An `double` called **topSpeed**

You need to add the following items:

- Constructor:
    - a `Car` constructor that is visible to other classes and accepts a `String` and a `double` and uses it to initialize the
      **maker** and **topSpeed** fields.

- Methods:

>     - a `getMaker` method which returns the value stored in the **maker** field

Reviewer: change this and any other "field" to "attribute"

>     - a `getTopSpeed` method which returns the value stored in the **topSpeed** field

Reviewer: we should consider adding a toString method here, since all the other section questions do
